Mathlib/Topology/Instances/AddCircle/DenseSubgroup.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2024 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Data.Real.Irrational
+import Mathlib.Topology.Instances.AddCircle
+import Mathlib.Topology.Algebra.Order.Archimedean
+
+/-!
+# Irrational rotation is minimal
+
+In this file we prove that the multiples of an irrational element of an `AddCircle` are dense.
+Moreover, an additive subgroup of the circle is dense
+iff it is not generated by a single element of finite additive order.
+-/
+
+open Set Filter
+open scoped Pointwise Topology
+
+/-- The additive subgroup of real numbers generated by `a` and `b` is dense
+iff `a / b` is an irrational number.
+
+Here we rely on the fact that `a / 0 = 0` in Mathlib,
+so we don't need to add `b ≠ 0` as an assumption. -/
+theorem dense_addSubgroupClosure_pair_iff {a b : ℝ} :
+    Dense (AddSubgroup.closure {a, b} : Set ℝ) ↔ Irrational (a / b) := by
```

Comment:
what do you think of stating this in terms of `Real.log`, the multiplicative version, and Circle below?

Reply:
This would add quite a few imports to the file. We can do it in another file in a different PR.

P.S.: IMHO, we have too many ways to talk about a circle in Mathlib, but I don't know a good way out of this situation.

Reply:
That sounds good to me! Could you please leave a TODO regarding this here or in the implementation notes?

Reply:
If you already have a specific statement in mind, could you please push a TODO with this statement?

Reply:
@pechersky What are `a` and `b` in your TODO? We definitely can have a version with `IsOfFinOrder` and `Subgroup.zpowers`.

Reply:
Ah, I was mistaken that we had a multiplicative version of `AddCircle`. I see now there is a TODO on the AddCircle file that `TODO: Exponential equivalence to Circle`